From e01ccb1f50492054732c4234a963832581a1aa8c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michael=20M=C3=BCller?= <michael@fds-team.de>
Date: Sun, 19 Feb 2017 17:38:00 +0100
Subject: wined3d: Set all default state values to zero.

This patch is necessary as some of the values contain random data which is
especially dangerous when the CS has its own state and the values differ.
The patch fixes geometry errors in the second level of FarCry and color
keying in the Motoracer 2.
---
 dlls/wined3d/stateblock.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/dlls/wined3d/stateblock.c b/dlls/wined3d/stateblock.c
index 463ccc91561..263b200c762 100644
--- a/dlls/wined3d/stateblock.c
+++ b/dlls/wined3d/stateblock.c
@@ -1358,6 +1358,9 @@ HRESULT state_init(struct wined3d_state *state, const struct wined3d_gl_info *gl
 {
     unsigned int i;
 
+    if (flags & WINED3D_STATE_INIT_DEFAULT)
+        memset(state, 0, sizeof(*state));
+
     state->flags = flags;
 
     for (i = 0; i < LIGHTMAP_SIZE; i++)
-- 
2.11.0

